pandas/tests/arithmetic/test_datetime64.py

```diff
@@ -1056,6 +1057,33 @@ def test_dt64arr_add_sub_parr(
         )
         assert_invalid_addsub_type(dtarr, parr, msg)
 
+    def test_timestamp_addsub_time_dtype_raises(self, box_with_array):
+        # https://github.com/pandas-dev/pandas/issues/10329
+        obj1 = pd.date_range("2012-01-01", periods=3)
+        obj2 = [time(i, i, i) for i in range(3)]
+
+        obj1 = tm.box_expected(obj1, box_with_array)
+        obj2 = tm.box_expected(obj2, box_with_array)
+
+        msg = (
+            r"unsupported operand type\(s\) for -: "
+            "'(Timestamp|DatetimeArray)' and 'datetime.time'"
+        )
+
+        with pytest.raises(TypeError, match=msg):
+            with warnings.catch_warnings(record=True):
+                # pandas.errors.PerformanceWarning: Non-vectorized DateOffset being
+                # applied to Series or DatetimeIndex
+                # we aren't testing that here, so ignore.
+                warnings.simplefilter("ignore", PerformanceWarning)
+                # sub
+                obj1 - obj2
+                obj2 - obj1
+
+                # add
+                obj1 + obj2
+                obj2 + obj1
```

Comment:
should we be testing that all four of these ops raise TypeError?

Reply:
Yea this isn't very clear as is whether one or all of these raise a TypeError - can these be split into different contexts?

Reply:
Or alternately you can parametrize add, radd, sub, rsub

Reply:
> Or alternately you can parametrize add, radd, sub, rsub

I can do that, is that's the preferred way?

Reply:
dealer's choice.  i tend not to parametrize over add/radd/sub/rsub since it repeats all the setup code for for what i see as little gain, but on a case-by-case basis if it improves clarity, go for it.